src/utils.js

```diff
@@ -44,12 +43,41 @@ export function setNextLinks(units) {
  */
 export function applyConfig({ obj, config = {}, callBuild = false }) {
     const { functions = {}, attributes = {} } = config;
+    // eslint-disable-next-line no-restricted-syntax
     for (const [func, args] of Object.entries(functions)) {
-        obj[func] ? args ? obj[func](args) : obj[func]() : null;
+        // eslint-disable-next-line no-nested-ternary
+        obj[func] ? (args ? obj[func](args) : obj[func]()) : null;
     }
-    const modified = (callBuild) ? obj.build() : obj;
+    const modified = callBuild ? obj.build() : obj;
+    // eslint-disable-next-line no-restricted-syntax
     for (const [key, values] of Object.entries(attributes)) {
         modified[key] = values;
     }
     return modified;
 }
+
+/**
+ * @summary Safely extract unit object from subworkflow data
+ * @param subworkflowData {Object} subworkflow data
+ * @param index {number} index of subworkflow unit
+ * @param type {string} type of subworkflow unit
+ * @returns {Object|null} subworkflow unit object (not a unit class instance!)
+ */
+export function findUnit({ subworkflowData, index, type }) {
+    const unit = subworkflowData.units[index];
+    if (unit.type !== type) throw new Error("findUnit() error: unit type does not match!");
+    return unit;
+}
+
+/**
+ * @summary Apply the patch operation to a subworkflow unit
+ * @param unit {Object} subworkflow unit object
+ * @param operations {Object} patch instructions
+ * @todo Add more operations, e.g. setName, etc.
+ */
+export function applyPatch({ unit, operations }) {
+    const { setProp } = operations;
+    if (setProp) {
+        Object.entries(operations.setProp).forEach(([key, value]) => (unit.config[key] = value));
```

Comment:
Don't we already have "config" field to do exactly this?

Reply:
`applyConfig` can also be used, although it introduces an additional level of nesting as the `attributes` is required to set attributes. I adjusted it so that `applyConfig` is used.

Reply:
also to answer your question:
not exactly, the current usage of the `config` applies only to workflows or subworkflows. `config` for units of subworkflow cannot be set from the workflow asset